controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -0,0 +1,98 @@
+			// the ToolChainClusters are created or deleted. We don't have to care about updates, the mere existence of the
+			// ToolchainCluster is enough for us.
+			&source.Kind{Type: &toolchainv1alpha1.ToolchainCluster{}},
+			handler.EnqueueRequestsFromMapFunc(MapToolChainClusterToSpaceProvisionerConfigs(ctx, r.Client)),
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(event.CreateEvent) bool {
+					return true
+				},
+				DeleteFunc: func(event.DeleteEvent) bool {
+					return true
+				},
+			},
+			)).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaceprovisionerconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaceprovisionerconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=toolchainclusters,verbs=get;list;watch
+
+// Reconcile ensures that SpaceProvisionerConfig is valid and points to an existing ToolchainCluster.
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	spaceProvisionerConfig := &toolchainv1alpha1.SpaceProvisionerConfig{}
+	if err := r.Client.Get(ctx, request.NamespacedName, spaceProvisionerConfig); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("SpaceProvisionerConfig not found anymore")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get SpaceProvisionerConfig: %w", err)
+	}
+
+	if util.IsBeingDeleted(spaceProvisionerConfig) {
+		logger.Info("SpaceProvisionerConfig is being deleted - skipping reconciliation")
+		return ctrl.Result{}, nil
+	}
+
+	// check that there exists a ToolchainCluster CR in the same namespace
+	toolchainCluster := &toolchainv1alpha1.ToolchainCluster{}
+	toolchainClusterKey := runtimeclient.ObjectKey{Name: spaceProvisionerConfig.Spec.ToolchainCluster, Namespace: spaceProvisionerConfig.Namespace}
+	toolchainPresent := corev1.ConditionTrue
+	toolchainPresenceReason := toolchainv1alpha1.SpaceProvisionerConfigValidReason
+	if err := r.Client.Get(ctx, toolchainClusterKey, toolchainCluster); err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{}, fmt.Errorf("failed to get ToolchainCluster: %w", err)
+		}
+		toolchainPresenceReason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason
+		toolchainPresent = corev1.ConditionFalse
+	}
+
+	spaceProvisionerConfig.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spaceProvisionerConfig.Status.Conditions,
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: toolchainPresent,
+			Reason: toolchainPresenceReason,
+		})
+
+	if err := r.Client.Status().Update(ctx, spaceProvisionerConfig); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to update the SpaceProvisionerConfig status: %w", err)
```

Comment:
I would rather use `Wrap` function:
```go
errs "github.com/pkg/errors"

errs.Wrapf(err, "failed to update the SpaceProvisionerConfig status")
```

Reply:
`fmt.Errorf` is recommended in the `errors` package documentation (https://pkg.go.dev/errors@go1.21.6#pkg-overview). What are the advantages of using https://github.com/pkg/errors (btw that repo is archived and therefore, I assume, no longer maintained)?

Reply:
whoops, good catch :grimacing: :man_facepalming: 
yup, let's keep the `fmt.Errorf` there then
sorry for confusion

Reply:
created issue for changing that in the rest of our code https://issues.redhat.com/browse/KSPACE-32